### ISSUE TITLE
Help, A tutorial? 

### ISSUE THREAD

**Issue by AlejandroC90** [rubenlagus/TelegramBotsExample#12]
I know this is not an issue, but I need help to create a bot, I don't even know how start.
I put the bot token and run Main.java and just errors 

"Oct 26, 2015 2:15:56 PM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.io.FileNotFoundException) caught when processing request to {s}->https://api.telegram.org:443: path/to/my/certkey.pem (No such file or directory)
Oct 26, 2015 2:15:56 PM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {s}->https://api.telegram.org:443
Oct 26, 2015 2:15:56 PM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.io.FileNotFoundException) caught when processing request to {s}->https://api.telegram.org:443: path/to/my/certkey.pem (No such file or directory)
Oct 26, 2015 2:15:56 PM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {s}->https://api.telegram.org:443
Oct 26, 2015 2:15:56 PM org.apache.http.impl.execchain.RetryExec execute
INFO: I/O exception (java.io.FileNotFoundException) caught when processing request to {s}->https://api.telegram.org:443: path/to/my/certkey.pem (No such file or directory)
Oct 26, 2015 2:15:56 PM org.apache.http.impl.execchain.RetryExec execute
INFO: Retrying request to {s}->https://api.telegram.org:443
Exception in thread "main" java.lang.NullPointerException
    at org.telegram.database.ConectionDB.checkVersion(ConectionDB.java:72)
    at org.telegram.database.DatabaseManager.<init>(DatabaseManager.java:38)
    at org.telegram.database.DatabaseManager.getInstance(DatabaseManager.java:55)
    at org.telegram.updatesreceivers.UpdatesThread.<init>(UpdatesThread.java:47)
    at org.telegram.updateshandlers.FilesHandlers.<init>(FilesHandlers.java:48)
    at org.telegram.Main.initBots(Main.java:37)
    at org.telegram.Main.main(Main.java:24)
"
Don't even know what to do next, or why is the error, I would like to know if there is even a simple tutorial using this library.

**Comment by rubenlagus** [comment]
Just disable the "useWebhook" field in BuildVars and it should work (this field is only useful if you want to use webhook option)

In addition, make sure you filled all fields in BuildVars before using the examples bots, otherwise they won't work. For the stacktrace, it seems you missed also the the database connection.

**Comment by Cs4r** [comment]
Hi folks.

I am having the same issues that @AlejandroC90 mentioned, in my humble opinion a quick start guide for this project is missing. 

I offer myself as a volunteer, even though I'd need some initial help for making the project run first.

What do you think @rubenlagus ??

**Comment by rubenlagus** [comment]
@Cs4r Feel free to make it and add it to a pull request. I would make it, but I have no time for now, sorry. In any case, feel free to contact me if you need any small detail

**Comment by ghost** [comment]
Hi, so is there going to be a tutorial for this thing?

**Comment by Cs4r** [comment]
@nelsongoh there isn't such a thing. I gave up because I didn't get enough help from the author and also found a good alternative written in Python. It seems that the project became obsolete.

**Comment by rubenlagus** [comment]
@nelsongoh @Cs4r I'm working on it, but first I want to completely split the library from the examples bots (I'm adding only library to https://github.com/rubenlagus/TelegramBots), once it is complete, I will add several things to that projects:
1. Documentation
2. Maven usage
3. Compiled jars

Please, keep in mind that the new repo  is under construction.

**Comment by Cs4r** [comment]
wanna help? Is there any to-do list we can take a look for giving you a hand?

**Comment by rubenlagus** [comment]
@Cs4r It would be great if you can give it a test... I will add docs step by step:  https://jitpack.io/#rubenlagus/TelegramBots/v1.0

By the way, dev branch here is already using this library... And I'm updating my bots asap

**Comment by Cs4r** [comment]
I will take a look during this week-end. What do you want me to test exactly?

**Comment by rubenlagus** [comment]
@Cs4r My bots have been running for a week with this new version and long polling mode, so I guess there are not big issues in it. In any case, anything you can test is welcome, just to ensure I didn't miss anything. Thanks a log

**Comment by smartDev22** [comment]
I'm sorry, could you tell, I have so any typing errors:
`"C:\Program Files\Java\jdk1.8.0_73\bin\java" -Didea.launcher.port=7534 "-Didea.launcher.bin.path=C:\Program Files (x86)\JetBrains\IntelliJ IDEA Community Edition 15.0.3\bin" -Dfile.encoding=UTF-8 -classpath "C:\Program Files\Java\jdk1.8.0_73\jre\lib\charsets.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\deploy.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\access-bridge-64.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\cldrdata.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\dnsns.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\jaccess.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\jfxrt.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\localedata.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\nashorn.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\sunec.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\sunjce_provider.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\sunmscapi.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\sunpkcs11.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\ext\zipfs.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\javaws.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\jce.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\jfr.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\jfxswt.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\jsse.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\management-agent.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\plugin.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\resources.jar;C:\Program Files\Java\jdk1.8.0_73\jre\lib\rt.jar;F:\TelegramBotsExample-master\TelegramBotsExample-master\target\classes;C:\Users\Alex\.m2\repository\com\github\rubenlagus\TelegramBots\v1.0\TelegramBots-v1.0.jar;C:\Users\Alex\.m2\repository\org\glassfish\jersey\containers\jersey-container-grizzly2-http\2.22.1\jersey-container-grizzly2-http-2.22.1.jar;C:\Users\Alex\.m2\repository\org\glassfish\hk2\external\javax.inject\2.4.0-b31\javax.inject-2.4.0-b31.jar;C:\Users\Alex\.m2\repository\org\glassfish\grizzly\grizzly-http-server\2.3.23\grizzly-http-server-2.3.23.jar;C:\Users\Alex\.m2\repository\org\glassfish\grizzly\grizzly-http\2.3.23\grizzly-http-2.3.23.jar;C:\Users\Alex\.m2\repository\org\glassfish\grizzly\grizzly-framework\2.3.23\grizzly-framework-2.3.23.jar;C:\Users\Alex\.m2\repository\org\glassfish\jersey\core\jersey-common\2.22.1\jersey-common-2.22.1.jar;C:\Users\Alex\.m2\repository\javax\annotation\javax.annotation-api\1.2\javax.annotation-api-1.2.jar;C:\Users\Alex\.m2\repository\org\glassfish\jersey\bundles\repackaged\jersey-guava\2.22.1\jersey-guava-2.22.1.jar;C:\Users\Alex\.m2\repository\org\glassfish\hk2\hk2-api\2.4.0-b31\hk2-api-2.4.0-b31.jar;C:\Users\Alex\.m2\repository\org\glassfish\hk2\hk2-utils\2.4.0-b31\hk2-utils-2.4.0-b31.jar;C:\Users\Alex\.m2\repository\org\glassfish\hk2\external\aopalliance-repackaged\2.4.0-b31\aopalliance-repackaged-2.4.0-b31.jar;C:\Users\Alex\.m2\repository\org\glassfish\hk2\hk2-locator\2.4.0-b31\hk2-locator-2.4.0-b31.jar;C:\Users\Alex\.m2\repository\org\javassist\javassist\3.18.1-GA\javassist-3.18.1-GA.jar;C:\Users\Alex\.m2\repository\org\glassfish\hk2\osgi-resource-locator\1.0.1\osgi-resource-locator-1.0.1.jar;C:\Users\Alex\.m2\repository\org\glassfish\jersey\core\jersey-server\2.22.1\jersey-server-2.22.1.jar;C:\Users\Alex\.m2\repository\org\glassfish\jersey\core\jersey-client\2.22.1\jersey-client-2.22.1.jar;C:\Users\Alex\.m2\repository\org\glassfish\jersey\media\jersey-media-jaxb\2.22.1\jersey-media-jaxb-2.22.1.jar;C:\Users\Alex\.m2\repository\javax\validation\validation-api\1.1.0.Final\validation-api-1.1.0.Final.jar;C:\Users\Alex\.m2\repository\javax\ws\rs\javax.ws.rs-api\2.0.1\javax.ws.rs-api-2.0.1.jar;C:\Users\Alex\.m2\repository\org\glassfish\jersey\media\jersey-media-json-jackson\2.22.1\jersey-media-json-jackson-2.22.1.jar;C:\Users\Alex\.m2\repository\org\glassfish\jersey\ext\jersey-entity-filtering\2.22.1\jersey-entity-filtering-2.22.1.jar;C:\Users\Alex\.m2\repository\com\fasterxml\jackson\jaxrs\jackson-jaxrs-base\2.5.4\jackson-jaxrs-base-2.5.4.jar;C:\Users\Alex\.m2\repository\com\fasterxml\jackson\core\jackson-core\2.5.4\jackson-core-2.5.4.jar;C:\Users\Alex\.m2\repository\com\fasterxml\jackson\core\jackson-databind\2.5.4\jackson-databind-2.5.4.jar;C:\Users\Alex\.m2\repository\com\fasterxml\jackson\jaxrs\jackson-jaxrs-json-provider\2.5.4\jackson-jaxrs-json-provider-2.5.4.jar;C:\Users\Alex\.m2\repository\com\fasterxml\jackson\module\jackson-module-jaxb-annotations\2.5.4\jackson-module-jaxb-annotations-2.5.4.jar;C:\Users\Alex\.m2\repository\com\fasterxml\jackson\core\jackson-annotations\2.5.4\jackson-annotations-2.5.4.jar;C:\Users\Alex\.m2\repository\com\sun\jersey\jersey-bundle\1.19\jersey-bundle-1.19.jar;C:\Users\Alex\.m2\repository\javax\ws\rs\jsr311-api\1.1.1\jsr311-api-1.1.1.jar;C:\Users\Alex\.m2\repository\com\sun\jersey\jersey-grizzly2-servlet\1.19\jersey-grizzly2-servlet-1.19.jar;C:\Users\Alex\.m2\repository\org\glassfish\grizzly\grizzly-http-servlet\2.2.16\grizzly-http-servlet-2.2.16.jar;C:\Users\Alex\.m2\repository\com\sun\jersey\jersey-grizzly2\1.19\jersey-grizzly2-1.19.jar;C:\Users\Alex\.m2\repository\com\sun\jersey\jersey-server\1.19\jersey-server-1.19.jar;C:\Users\Alex\.m2\repository\com\sun\jersey\jersey-core\1.19\jersey-core-1.19.jar;C:\Users\Alex\.m2\repository\javax\servlet\javax.servlet-api\3.0.1\javax.servlet-api-3.0.1.jar;C:\Users\Alex\.m2\repository\com\sun\jersey\jersey-servlet\1.19\jersey-servlet-1.19.jar;C:\Users\Alex\.m2\repository\org\json\json\20151123\json-20151123.jar;C:\Users\Alex\.m2\repository\commons-io\commons-io\2.4\commons-io-2.4.jar;C:\Users\Alex\.m2\repository\org\apache\httpcomponents\httpclient\4.5.1\httpclient-4.5.1.jar;C:\Users\Alex\.m2\repository\org\apache\httpcomponents\httpcore\4.4.3\httpcore-4.4.3.jar;C:\Users\Alex\.m2\repository\commons-logging\commons-logging\1.2\commons-logging-1.2.jar;C:\Users\Alex\.m2\repository\commons-codec\commons-codec\1.9\commons-codec-1.9.jar;C:\Users\Alex\.m2\repository\org\apache\httpcomponents\httpmime\4.5.1\httpmime-4.5.1.jar;C:\Users\Alex\.m2\repository\mysql\mysql-connector-java\5.1.38\mysql-connector-java-5.1.38.jar;C:\Users\Alex\.m2\repository\org\jsoup\jsoup\1.8.3\jsoup-1.8.3.jar;C:\Program Files (x86)\JetBrains\IntelliJ IDEA Community Edition 15.0.3\lib\idea_rt.jar" com.intellij.rt.execution.application.AppMain org.telegram.Main
Exception in thread "main" java.lang.IllegalArgumentException: Illegal character in path at index 28: https://api.telegram.org/bot<token>/setwebhook
    at java.net.URI.create(URI.java:852)
    at org.apache.http.client.methods.HttpPost.<init>(HttpPost.java:76)
    at org.telegram.telegrambots.TelegramBotsApi.setWebhook(TelegramBotsApi.java:139)
    at org.telegram.telegrambots.TelegramBotsApi.setWebhook(TelegramBotsApi.java:111)
    at org.telegram.telegrambots.TelegramBotsApi.registerBot(TelegramBotsApi.java:88)
    at org.telegram.Main.main(Main.java:21)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
Caused by: java.net.URISyntaxException: Illegal character in path at index 28: https://api.telegram.org/bot<token>/setwebhook
    at java.net.URI$Parser.fail(URI.java:2848)
    at java.net.URI$Parser.checkChars(URI.java:3021)
    at java.net.URI$Parser.parseHierarchical(URI.java:3105)
    at java.net.URI$Parser.parse(URI.java:3053)
    at java.net.URI.<init>(URI.java:588)
    at java.net.URI.create(URI.java:850)
    ... 10 more
`
Here file BuildVar.java:

`
public class BuildVars {
    public static final Boolean debug = true;
    //public static final Boolean useWebHook = false;
    public static final int PORT = 8443;
    public static final String EXTERNALWEBHOOKURL = "your-external-url:" + PORT;
    public static final String INTERNALWEBHOOKURL = "your-internal-url:" + PORT;
    public static final String pathToCertificatePublicKey = "path/to/my/certkey.pem";
    public static final String certificatePublicKeyFileName = "certkey.pem";

```
public static final String OPENWEATHERAPIKEY = "my api";

public static final String DirectionsApiKey = "my api";

public static final String TRANSIFEXUSER = "<transifex-user>";
public static final String TRANSIFEXPASSWORD = "<transifex-password>";

public static final String pathToLogs = "./";

public static final String linkDB = "jdbc:mysql://localhost:3306/tg?useUnicode=true&characterEncoding=UTF-8";
public static final String controllerDB = "com.mysql.jdbc.Driver";
public static final String userDB = "root";
public static final String password = "";
```

}`

sorry for my English, I use a translator

**Comment by rubenlagus** [comment]
@sitecreate it seems you are not filling [BotConfig](https://github.com/rubenlagus/TelegramBotsExample/blob/master/src/main/java/org/telegram/BotConfig.java) correctly, please have a look.

**Comment by smartDev22** [comment]
Now is such a sorry mistake:

> Information:Using javac 1.8.0_73 to compile java sources
> Information:java: Errors occurred while compiling module 'main'
> Information:23.02.2016 15:21 - Compilation completed with 100 errors and 0 warnings in 1s 322ms
> F:\TelegramBotsExample-master\TelegramBotsExample-master\src\main\java\org\telegram\updateshandlers\FilesHandlers.java
> Error:(9, 33) java: package org.telegram.telegrambots does not exist
> Error:(10, 45) java: package org.telegram.telegrambots.api.methods does not exist
> Error:(11, 45) java: package org.telegram.telegrambots.api.methods does not exist
> Error:(12, 45) java: package org.telegram.telegrambots.api.objects does not exist
> Error:(13, 45) java: package org.telegram.telegrambots.api.objects does not exist
> Error:(14, 45) java: package org.telegram.telegrambots.api.objects does not exist
> Error:(15, 45) java: package org.telegram.telegrambots.api.objects does not exist
> Error:(16, 38) java: package org.telegram.telegrambots.bots does not exist
> Error:(32, 36) java: cannot find symbol
>   symbol: class TelegramLongPollingBot
> Error:(45, 34) java: cannot find symbol
>   symbol:   class Update
>   location: class org.telegram.updateshandlers.FilesHandlers
> Error:(58, 35) java: cannot find symbol
>   symbol:   class Update
>   location: class org.telegram.updateshandlers.FilesHandlers
> Error:(106, 32) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.FilesHandlers
> Error:(130, 34) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.FilesHandlers
> Error:(139, 51) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.FilesHandlers
> Error:(166, 48) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.FilesHandlers
> Error:(185, 34) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.FilesHandlers
> Error:(197, 34) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.FilesHandlers
> Error:(209, 34) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.FilesHandlers
> Error:(224, 40) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.FilesHandlers
> Error:(242, 39) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.FilesHandlers
> Error:(267, 37) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.FilesHandlers
> F:\TelegramBotsExample-master\TelegramBotsExample-master\src\main\java\org\telegram\services\RaeService.java
> Error:(3, 23) java: package org.apache.http does not exist
> Error:(4, 38) java: package org.apache.http.client.methods does not exist
> Error:(5, 38) java: package org.apache.http.client.methods does not exist
> Error:(6, 32) java: package org.apache.http.conn.ssl does not exist
> Error:(7, 30) java: package org.apache.http.entity does not exist
> Error:(8, 35) java: package org.apache.http.impl.client does not exist
> Error:(9, 35) java: package org.apache.http.impl.client does not exist
> Error:(10, 28) java: package org.apache.http.util does not exist
> Error:(11, 17) java: package org.jsoup does not exist
> Error:(12, 23) java: package org.jsoup.nodes does not exist
> Error:(13, 23) java: package org.jsoup.nodes does not exist
> Error:(14, 24) java: package org.jsoup.select does not exist
> Error:(145, 54) java: cannot find symbol
>   symbol:   class Elements
>   location: class org.telegram.services.RaeService
> Error:(182, 53) java: cannot find symbol
>   symbol:   class Element
>   location: class org.telegram.services.RaeService
> F:\TelegramBotsExample-master\TelegramBotsExample-master\src\main\java\org\telegram\Main.java
> Error:(4, 33) java: package org.telegram.telegrambots does not exist
> Error:(5, 33) java: package org.telegram.telegrambots does not exist
> F:\TelegramBotsExample-master\TelegramBotsExample-master\src\main\java\org\telegram\updateshandlers\DirectionsHandlers.java
> Error:(3, 16) java: package org.json does not exist
> Error:(10, 33) java: package org.telegram.telegrambots does not exist
> Error:(11, 45) java: package org.telegram.telegrambots.api.methods does not exist
> Error:(12, 45) java: package org.telegram.telegrambots.api.methods does not exist
> Error:(13, 1) java: package org.telegram.telegrambots.api.objects does not exist
> Error:(14, 38) java: package org.telegram.telegrambots.bots does not exist
> Error:(15, 49) java: package org.telegram.telegrambots.updateshandlers does not exist
> Error:(30, 41) java: cannot find symbol
>   symbol: class TelegramLongPollingBot
> Error:(44, 34) java: cannot find symbol
>   symbol:   class Update
>   location: class org.telegram.updateshandlers.DirectionsHandlers
> Error:(57, 35) java: cannot find symbol
>   symbol:   class Update
>   location: class org.telegram.updateshandlers.DirectionsHandlers
> Error:(101, 40) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.DirectionsHandlers
> Error:(138, 35) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.DirectionsHandlers
> Error:(172, 34) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.DirectionsHandlers
> Error:(186, 43) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.DirectionsHandlers
> Error:(220, 39) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.DirectionsHandlers
> Error:(245, 37) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.DirectionsHandlers
> F:\TelegramBotsExample-master\TelegramBotsExample-master\src\main\java\org\telegram\services\DirectionsService.java
> Error:(3, 23) java: package org.apache.http does not exist
> Error:(4, 23) java: package org.apache.http does not exist
> Error:(5, 30) java: package org.apache.http.client does not exist
> Error:(6, 38) java: package org.apache.http.client.methods does not exist
> Error:(7, 32) java: package org.apache.http.conn.ssl does not exist
> Error:(8, 30) java: package org.apache.http.entity does not exist
> Error:(9, 35) java: package org.apache.http.impl.client does not exist
> Error:(10, 28) java: package org.apache.http.util does not exist
> Error:(11, 16) java: package org.json does not exist
> Error:(12, 16) java: package org.json does not exist
> Error:(13, 17) java: package org.jsoup does not exist
> Error:(110, 45) java: cannot find symbol
>   symbol:   class JSONArray
>   location: class org.telegram.services.DirectionsService
> Error:(127, 40) java: cannot find symbol
>   symbol:   class JSONObject
>   location: class org.telegram.services.DirectionsService
> F:\TelegramBotsExample-master\TelegramBotsExample-master\src\main\java\org\telegram\services\WeatherService.java
> Error:(3, 23) java: package org.apache.http does not exist
> Error:(4, 38) java: package org.apache.http.client.methods does not exist
> Error:(5, 38) java: package org.apache.http.client.methods does not exist
> Error:(6, 32) java: package org.apache.http.conn.ssl does not exist
> Error:(7, 30) java: package org.apache.http.entity does not exist
> Error:(8, 35) java: package org.apache.http.impl.client does not exist
> Error:(9, 35) java: package org.apache.http.impl.client does not exist
> Error:(10, 28) java: package org.apache.http.util does not exist
> Error:(11, 16) java: package org.json does not exist
> Error:(271, 50) java: cannot find symbol
>   symbol:   class JSONObject
>   location: class org.telegram.services.WeatherService
> Error:(293, 50) java: cannot find symbol
>   symbol:   class JSONObject
>   location: class org.telegram.services.WeatherService
> Error:(308, 55) java: cannot find symbol
>   symbol:   class JSONObject
>   location: class org.telegram.services.WeatherService
> Error:(359, 38) java: cannot find symbol
>   symbol:   class JSONObject
>   location: class org.telegram.services.WeatherService
> F:\TelegramBotsExample-master\TelegramBotsExample-master\src\main\java\org\telegram\updateshandlers\ChannelHandlers.java
> Error:(5, 33) java: package org.telegram.telegrambots does not exist
> Error:(6, 45) java: package org.telegram.telegrambots.api.methods does not exist
> Error:(7, 45) java: package org.telegram.telegrambots.api.objects does not exist
> Error:(8, 45) java: package org.telegram.telegrambots.api.objects does not exist
> Error:(9, 45) java: package org.telegram.telegrambots.api.objects does not exist
> Error:(10, 45) java: package org.telegram.telegrambots.api.objects does not exist
> Error:(11, 38) java: package org.telegram.telegrambots.bots does not exist
> Error:(23, 38) java: cannot find symbol
>   symbol: class TelegramLongPollingBot
> Error:(38, 34) java: cannot find symbol
>   symbol:   class Update
>   location: class org.telegram.updateshandlers.ChannelHandlers
> Error:(63, 40) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.ChannelHandlers
> Error:(76, 42) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.ChannelHandlers
> Error:(95, 56) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.ChannelHandlers
> Error:(110, 35) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.ChannelHandlers
> Error:(126, 56) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.ChannelHandlers
> Error:(126, 20) java: cannot find symbol
>   symbol:   class SendMessage
>   location: class org.telegram.updateshandlers.ChannelHandlers
> Error:(142, 56) java: cannot find symbol
>   symbol:   class Message
>   location: class org.telegram.updateshandlers.ChannelHandlers
> Error:(142, 20) java: cannot find symbol
>   symbol:   class SendMessage
>   location: class org.telegram.updateshandlers.ChannelHandlers
> Error:(152, 68) java: cannot find symbol
>   symbol:   class ReplyKeyboardMarkup
>   location: class org.telegram.updateshandlers.ChannelHandlers
> F:\TelegramBotsExample-master\TelegramBotsExample-master\src\main\java\org\telegram\updateshandlers\RaeHandlers.java
> Error:(6, 33) java: package org.telegram.telegrambots does not exist
> Error:(7, 45) java: package org.telegram.telegrambots.api.methods does not exist
> Error:(8, 45) java: package org.telegram.telegrambots.api.methods does not exist

**Comment by smartDev22** [comment]
I understand it is not enough packages, but they are not in committees

**Comment by Clevero** [comment]
I quickly wrote a [tutorial](https://gist.github.com/Clevero/f32b35139f6c900cdc01). Can someone please give some feedback for it?

**Comment by rgf2004** [comment]
@Clevero , it worked fine with me, thanks for your help your really made my day by understanding how  i can use this library, :)

**Comment by rubenlagus** [comment]
@Clevero It looks great, if you make a pull request, I'll merge asap

**Comment by Lorentz18** [comment]
I'm sorry to hijack this 3d but i have a problem, first i'm a total noob in github and probably i'm fucking up somewhere but i followed @Clevero tutorial and all goes well till i have to locate BotConfig.java which is not existent in any directory of the imported project, furthermore i cannot find any file labelled as main.java nor buildvars.java, both mentioned in the tutorial. What am i doing wrong? Thank you
Edit- I realized those files exist in the telegram bots example but not in the download linked in the tutorial, thanks anyway :+1:

**Comment by rubenlagus** [comment]
Closed via #19
